packages/app-project/src/screens/ClassifyPage/components/ClassifierWrapper/ClassifierWrapper.js

```diff
@@ -71,26 +75,21 @@ export default function ClassifierWrapper({
     If the page URL contains a subject ID, update that ID when the classification subject changes.
     Subject page URLs can be either `/classify/workflow/{workflowID}/subject/{subjectID}`
     or `/classify/workflow/{workflowID}/subject-set/{subjectSetID}/subject/{subjectID}`.
+    Example: Subject Set Progress Banner arrow buttons
   */
-  const replaceRoute = router.replace
   let baseURL = `/${ownerSlug}/${projectSlug}/classify/workflow/${workflowID}`
   if (subjectSetID) {
     baseURL = `${baseURL}/subject-set/${subjectSetID}`
   }
-  let subjectPageURL = null
-  let subjectRouteChanged = false
-  if (subjectID) {
-    subjectPageURL = `${baseURL}/subject/${classifierSubjectID}`
-    subjectRouteChanged = router?.query.subjectID !== classifierSubjectID
-  }
 
-  useEffect(function updatePageURL() {
-    if (subjectPageURL && subjectRouteChanged) {
+  if (router.query.subjectID && classifierSubjectID) {
+    const subjectPageURL = `${baseURL}/subject/${classifierSubjectID}`
+    /* True when the classifier subject changed, but new id is not yet reflected in the url */
+    if (router?.query.subjectID !== classifierSubjectID) {
       const href = addQueryParams(subjectPageURL)
-      const as = href
-      replaceRoute(href, as, { shallow: true })
+      router.push(href, href, { shallow: true })
```

Comment:
```suggestion
      router.replace(href, href, { shallow: true })
```


I think this should replace the current history entry, rather than appending a new entry.